### DISCUSSION TITLE
Verifying bundle's binary compatibility?

### DISCUSSION THREAD

**Discussion by lppedd** [eclipse-tycho/tycho#3524] [Q&A]
What's the preferred way to verify binary compatibility of bundles' classes via bytecode?  
I'd like to dump the binary signatures out of class files to create a baseline, and than compare it when necessary.

I see there is a `p2-metadata` mojo that uses a validator, however I don't see any readme on how to use it.

**Comment by laeubi** [comment]
You should use https://tycho.eclipseprojects.io/doc/latest/tycho-baseline-plugin/verify-mojo.html for example m2e is using that.

**Reply by lppedd** [reply]
@laeubi Thanks. Does that mojo verify the class/method signatures, or just the manifests for version changes (as the documentation suggests)?
